subsys/debug/sysview/SEGGER_SYSVIEW_Zephyr.c

```diff
@@ -0,0 +1,197 @@
+}
+
+void sys_trace_thread_switched_in(void)
+{
+	struct k_thread *thread;
+
+	thread = k_current_get();
+
+	if (is_idle_thread(thread)) {
+		SEGGER_SYSVIEW_OnIdle();
+	} else {
+		SEGGER_SYSVIEW_OnTaskStartExec((u32_t)(uintptr_t)thread);
+	}
+}
+
+void sys_trace_thread_switched_out(void)
+{
+	SEGGER_SYSVIEW_OnTaskStopExec();
+}
+
+void sys_trace_isr_enter(void)
+{
+	SEGGER_SYSVIEW_RecordEnterISR();
+}
+
+void sys_trace_isr_exit(void)
+{
+	SEGGER_SYSVIEW_RecordExitISR();
+}
+
+
+void sys_trace_isr_exit_to_scheduler(void)
+{
+	SEGGER_SYSVIEW_RecordExitISRToScheduler();
+}
+
+void sys_trace_thread_priority_get(struct k_thread *thread)
+{
+}
+
+void sys_trace_thread_priority_set(struct k_thread *thread)
+{
+}
+
+void sys_trace_thread_create(struct k_thread *thread)
+{
+	SEGGER_SYSVIEW_OnTaskCreate((u32_t)(uintptr_t)thread);
+	sys_trace_thread_info(thread);
+}
+
+void sys_trace_thread_info(struct k_thread *thread)
+{
+	char name[20];
+
+	if (thread->name == NULL) {
+		snprintk(name, sizeof(name), "T%xE%x", (uintptr_t)thread,
+			 (uintptr_t)&thread->entry);
+	} else {
+		snprintk(name, sizeof(name), "%s", thread->name);
+	}
+
+	SEGGER_SYSVIEW_TASKINFO Info;
+
+	Info.TaskID = (u32_t)(uintptr_t)thread;
+	Info.sName = name;
+	Info.Prio = thread->base.prio;
+	Info.StackBase = thread->stack_info.size;
+	Info.StackSize = thread->stack_info.start;
+	SEGGER_SYSVIEW_SendTaskInfo(&Info);
+}
+
+void sys_trace_thread_abort(struct k_thread *thread)
+{
+}
+
+void sys_trace_thread_cancel(struct k_thread *thread)
+{
+}
+
+void sys_trace_thread_suspend(struct k_thread *thread)
+{
+}
+
+void sys_trace_thread_resume(struct k_thread *thread)
+{
+}
+
+void sys_trace_thread_ready(struct k_thread *thread)
+{
+	SEGGER_SYSVIEW_OnTaskStartReady((u32_t)(uintptr_t)thread);
+}
+
+void sys_trace_thread_pend(struct k_thread *thread)
+{
+	SEGGER_SYSVIEW_OnTaskStopReady((u32_t)(uintptr_t)thread, 3 << 3);
+}
+
+void sys_trace_void(unsigned int id)
+{
+	SEGGER_SYSVIEW_RecordVoid(id);
+}
+
+void sys_trace_idle(void)
+{
+	SEGGER_SYSVIEW_OnIdle();
+}
+
+void sys_trace_end_call(unsigned int id)
+{
+	SEGGER_SYSVIEW_RecordEndCall(id);
+}
+
+u32_t sysview_get_timestamp(void)
+{
+	return k_cycle_get_32();
+}
+
+u32_t sysview_get_interrupt(void)
+{
+	return interrupt;
+}
+
+static void send_task_list_cb(void)
+{
+	struct k_thread *thread;
+
+	for (thread = _kernel.threads; thread; thread = thread->next_thread) {
+		char name[20];
+
+		if (is_idle_thread(thread)) {
+			continue;
+		}
+
+		if (thread->name == NULL) {
+			snprintk(name, sizeof(name), "T%xE%x", (uintptr_t)thread,
+				 (uintptr_t)&thread->entry);
+		} else {
+			snprintk(name, sizeof(name), "%s", thread->name);
+		}
+		SEGGER_SYSVIEW_SendTaskInfo(&(SEGGER_SYSVIEW_TASKINFO) {
+			.TaskID = (u32_t)(uintptr_t)thread,
+			.sName = name,
+			.StackSize = thread->stack_info.size,
+			.StackBase = thread->stack_info.start,
+			.Prio = thread->base.prio,
+		});
+	}
+}
+
+
+static U64 get_time_cb(void)
+{
+	return (U64)k_cycle_get_32();
```

Comment:
@carlescufi: This is another subsystem which calls for nrf_rtc_timer.c refactor, as current implementation of k_cycle_get_32() is too slow for logger/tracing usage.

Reply:
Right. Do we have an issue to fix that beyond the comments I saw in the logging subsystem PR?

Reply:
@nordic-krch: Could you please create an issue about the problem with `k_cycle_get_32()` that you found?

Reply:
It's about slowness of timestamp function on nrf series: #8798. Note that system timer based on counter.h should solve that issue.